cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -2536,6 +2553,42 @@ impl ShowWithRRU for Inst {
                 let rd = rd.show_rru(mb_rru);
                 format!("ldr {}, 8 ; b 12 ; data {:?} + {}", rd, name, offset)
             }
+            &Inst::LoadAddr { rd, ref mem } => match *mem {
+                MemArg::FPOffset(fp_off) => {
+                    let alu_op = if fp_off < 0 {
+                        ALUOp::Sub64
+                    } else {
+                        ALUOp::Add64
+                    };
+                    if let Some(imm12) = Imm12::maybe_from_u64(u64::try_from(fp_off.abs()).unwrap())
```

Comment:
Eventually I'd like to figure out a way to factor out the shared logic between `emit()` and `show_rru()` for the pseudo-instructions that lower to real instructions, without going back to the expensive edit-in-place-to-legalize paradigm; but for now this is OK, I think.

Reply:
Yeah, it would be nice to share this.